proposals/oci-support-updates.md

```diff
@@ -0,0 +1,95 @@
+
+```
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.cncf.helm.config.v1+json",
+    "digest": "sha256:8ec7c0f2f6860037c19b54c3cfbab48d9b4b21b485a93d87b64690fdb68c2111",
+    "size": 117
+  },
+  "layers": [
+    {
+      "mediaType": "application/tar+gzip",
+      "digest": "sha256:1b251d38cfe948dfc0a5745b7af5ca574ecb61e52aed10b19039db39af6e1617",
+      "size": 2487
+    },
+    {
+      "mediaType": "application/pgp-signature",
+      "digest": "sha256:3e207b409db364b595ba862cdc12be96dcdad8e36c59a03b7b3b61c946a5741a",
+      "size": 643
+    }
+  ]
+}
+
+```
+
+If `application/pgp-signature` is not appropriate based the contents of a `.prov` file (e.g. it does not meet the IANA requirements), a custom media type will be used such as `application/vnd.cncf.helm.provenance.v1`.
+
+## 3. Chart versions == OCI reference tags
+
+To keep things simple, the version of a chart will be 1-to-1 with the tag used on registry references. Arbitrary tags will not be supported for initial release. As time goes on, support for arbitrary tags may be reintroduced.
+
+This also means that tags are no longer necessary to be provided on the command-line in the form `<ref>:<tag>`. Some users feel strongly to keep this to match the UX of Docker and other tools, so perhaps this feature is kept and treated the same as using the `--version` flag. Thoughts?
+
+The latest sematic version is able to be determined from a registry by listing all tags. To specify a specific tag/version (or if a registry does not support tag listing), users can provide `--version <version>`.
+
+## 4. Chart names == OCI reference basenames
+
+Again, to keep things simple, the basename (the last segment of the URL path) on a registry reference should be equivalent to the chart name. 
+
+For example, given a chart with the name `pepper` and the version `1.2.3`, users may run a command such as the following:
+
+```
+$ helm push pepper/ oci://r.myreg.io/mycharts
+```
+
+which would result in the following reference:
+
+```
+oci://r.myreg.io/mycharts/pepper:1.2.3
+```
+
+By placing such restrictions on registry URLs and tags, Helm users are less likely to do "strange things" with charts in registries. Again, support for custom tags and basenames could be supported in the future via options such as `--oci-basename` after we first stabilize the feature set.
+
+## 5. `helm chart` is removed, integrated into rest of CLI
+
+Wherever possible, the subcommands provided by the new `helm chart` command should be integrated into existing Helm CLI commands.
+
+For example, instead of `helm chart save`, we may use something like `helm package --cache-oci` which would store it into the OCI cache. We may also choose to do away with the OCI cache entirely.
+
+`helm chart pull` should be baked into `helm pull`, `helm chart push` should be `helm push` (new). `helm chart remove` and `helm chart export` could potentially be removed if we no longer have an OCI cache. Thoughts on this?
```

Comment:
👍

Reply:
Should the cache be treated like the existing cache? Commands are not needed... though we may want to add a `helm cache` command in general.

Reply:
I'm wondering if we start with no cache, and add it later as a non-breaking change. There is some value there, as it will reduce overall transfer, but Helm charts are not nearly as big as Docker image layers where it really counts

Reply:
but yes, we can certainly cache without exposing commands at first to work with it